core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -58,6 +58,7 @@ case class SparkListenerTaskEnd(
 @DeveloperApi
 case class SparkListenerJobStart(
     jobId: Int,
+    time: Option[Long],
```

Comment:
I guess this is an option for backwards-compatibility reasons?  We definitely know the time when posting this event to the listener bus, so I think the right approach is to have time just be a regular `Long` and pass a dummy value (`-1`) when replaying JSON that's missing that field.